lib/Backend/IR.cpp

```diff
@@ -2858,129 +2858,155 @@ Instr::GetOrCreateContinueLabel(const bool isHelper)
     return label;
 }
 
-
-IR::RegOpnd * Instr::FindRegUseSrc(StackSym *sym, IR::Opnd* src)
+bool
+Instr::HasSymUseSrc(StackSym *sym, IR::Opnd* src)
 {
     if (!src)
     {
-        return nullptr;
+        return false;
     }
     if (src->IsRegOpnd())
     {
         RegOpnd *regOpnd = src->AsRegOpnd();
 
         if (regOpnd->m_sym == sym)
         {
-            return regOpnd;
+            return true;
         }
     }
     else if (src->IsIndirOpnd())
     {
         IR::IndirOpnd *indirOpnd = src->AsIndirOpnd();
+        if (indirOpnd->GetStackSym() == sym)
+        {
+            return true;
+        }
         RegOpnd * baseOpnd = indirOpnd->GetBaseOpnd();
         if (baseOpnd != nullptr && baseOpnd->m_sym == sym)
         {
-            return baseOpnd;
+            return true;
         }
         else if (indirOpnd->GetIndexOpnd() && indirOpnd->GetIndexOpnd()->m_sym == sym)
         {
-            return indirOpnd->GetIndexOpnd();
+            return true;
         }
     }
     else if (src->IsListOpnd())
     {
         IR::ListOpnd* list = src->AsListOpnd();
-        for (int i = 0; i < list->Count(); ++i)
+        if (list->GetStackSym() == sym)
         {
-            IR::RegOpnd* reg = FindRegUseSrc(sym, list->Item(i));
-            if (reg)
+            for (int i = 0; i < list->Count(); ++i)
             {
-                return reg;
+                if (HasSymUseSrc(sym, list->Item(i)))
+                {
+                    return true;
+                }
             }
         }
     }
-    return nullptr;
+    else if (src->IsSymOpnd())
+    {
+        SymOpnd* symOpnd = src->AsSymOpnd();
+        if (symOpnd->GetSym() == sym)
+        {
+            return true;
+        }
+        if (symOpnd->IsPropertySymOpnd())
+        {
+            PropertySymOpnd* propertySymOpnd = symOpnd->AsPropertySymOpnd();
+            if (propertySymOpnd->GetPropertySym()->m_stackSym == sym)
```

Comment:
`propertySymOpnd->GetPropertySym()->m_stackSym` and `propertySymOpnd->GetObjectSym()` returns the same stack sym. So the 2 checks are redondant